.github/workflows/deploy-docs.yml

```diff
@@ -22,6 +23,19 @@ env:
   GIT_EMAIL: '41898282+github-actions[bot]@users.noreply.github.com'
 
 jobs:
+  lint:
```

Comment:
What's the motivation of adding a linting job into this workflow? It runs separately of the others, if you want it to prevent the other jobs from running, then like the `add-version-to-docs` job, you should have `deploy` include a `needs` dependency on `lint`, thus it will run sequentially like `lint --> deploy --> add-version-to-docs` with a job failing preventing any subsequent jobs from running.

There's also "composite run steps" which allows for sharing `run` steps in a job across workflows, but in this case, just deferring to a shell script like `update-versions-json.sh` should be a good way to keep workflow duplicate jobs DRY and terse.

Alternatively, as suggested, we use a linting action. There are a few popular ones that bundle various linters and keep those up to date for us, we just supply any additional configuration if we need to. No need to pull the latest release to run like handled below in `run`.

I can see that this `lint` job is only being used against the single docs related script? You can use an `if` condition in that case so it's not as misleading with a job name like `Linting` when it's only for that specific file. But this could equally work well in the main lint workflow to identify the problem in PRs? (which don't deploy)